### DISCUSSION TITLE
The future of realm-kotlin?

### DISCUSSION THREAD

**Discussion by Meijuh** [realm/realm-kotlin#1829] [Q&A] [answered]
Device sync has been announced deprecated: https://www.mongodb.com/docs/atlas/app-services/sync/device-sync-deprecation/, and the realm-js repo already shows a deprecation warning. What is the future of realm-kotlin?

**Comment by nirinchev** [comment, marked as answer]
If you're using the local database, you can continue doing so for as long as you want. We're in the process of cleaning up the repo and establishing a `community` branch, which only has the local functionality (i.e. no Atlas Device Sync/App Services). While the MongoDB team will not be actively contributing to this project, we'd be happy to accept community contributions and are willing to assist with PR reviews/releases. If you're using Sync, the timelines communicated in the blog post apply (i.e. end-of-life in 2025) and it's best to reach out to your account executive or the support team if you need assistance with migrating your application.

**Reply by rodrigobdz** [reply]
@nirinchev Are there any plans for [`realm-kotlin`](https://github.com/realm/realm-kotlin) to sync the local database to Atlas without Device Sync? The deprecation page mentions a few [alternatives](https://www.mongodb.com/docs/atlas/app-services/sync/device-sync-deprecation/#alternative-solutions)—of which only [Parse](https://parseplatform.org/) seems to be open source. 

The issue seems that Parse is in an [uncertain state](https://github.com/parse-community/Parse-SDK-Android/issues/1100), waiting for a major overhaul since 2021.
